### PR TITLE
Scripts/Commands: New argument parsing class

### PR DESCRIPTION
**Pull request by Treeston**
New command argument parser. Proper description later, have to run now.

### CODE REVIEW

**Comment by Rochet2** [comment]
Related? https://github.com/TrinityCore/TrinityCore/pull/22350
Seems like the same thing, but different approach and maybe capabilities.

File: src/server/scripts/Commands/ArgsParserHyperlinks.hpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    static std::enable_if_t<std::is_integral_v<T> && std::is_signed_v<T>, bool> store(T& val, char const* pos, size_t len)
+    {
+        try { val = std::stoll(std::string(val, len)); }
+        catch (...) { return false; }
+        return true;
+    }
+};
+
+make_base_tag(creature, uint32);
+make_base_tag(creature_entry, uint32);
```
Comment (Warpten) [inline src/server/scripts/Commands/ArgsParserHyperlinks.hpp]: undef that

**Comment by Treeston** [comment]
just noticed the `_t` and `_v` helpers are technically in c++17, are there any relevant compilers that still lack them?

**Comment by Shauren** [comment]
Your args dont have names :P ref https://gist.github.com/Shauren/fe18d8b786bf4732bf2043ec7b76435c

File: src/server/game/Chat/ChatCommandArgs.hpp [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+|*   returns true iff value is present and of specified type                            *|
+|*                                                                                      *|
+\****************************************************************************************/
+class PlainInteger
+{
+    public:
+        template<typename T = int64>
+        T get() const
+        {
+            static_assert(std::is_integral_v<T>, "Attempt to extract PlainInteger using non-integral type");
```
Comment (illfated) [inline src/server/game/Chat/ChatCommandArgs.hpp]: Recurring error in travis-ci:
```cpp
src/server/game/Chat/ChatCommandArgs.hpp:63:32:
 fatal error: no template named 'is_integral_v' in namespace 'std'; did you mean 'is_integral'?
```
is this intended, that you are planning to add code to define `is_integral_v` somewhere?

**Review by Shauren — APPROVED** [review]

**Pull request merged**